library/types/test/hostname_test.rb

```diff
@@ -10,30 +10,49 @@
 
   let(:etc_hostname) { "etc.hostname.cz" }
   let(:cmd_hostname) { "cmd.hostname.cz" }
+  let(:hostname)     { Yast::Hostname }
+
+  def allow_execute_hostname(result)
+    ret = { "stdout" => result, "exit" => 0 } if result
 
-  it "returns output of hostname --fqdn if available" do
     allow(Yast::SCR)
       .to receive(:Execute)
       .with(path(".target.bash_output"), "hostname --fqdn")
-      .and_return("stdout" => cmd_hostname, "exit" => 0)
-
-    expect(Yast::Hostname.CurrentFQ).to eq cmd_hostname
+      .and_return(ret)
   end
 
-  it "reads /etc/hostname when hostname --fqdn fails" do
-    allow(Yast::SCR)
-      .to receive(:Execute)
-      .with(path(".target.bash_output"), "hostname --fqdn")
-      .and_return(nil)
+  def allow_read_hostname(result)
     allow(Yast::SCR)
       .to receive(:Read)
       .with(path(".target.string"), "/etc/hostname")
-      .and_return(etc_hostname)
+      .and_return(result)
+  end
+
+  it "returns output of hostname --fqdn if available" do
+    allow_execute_hostname(cmd_hostname)
+
+    expect(hostname.CurrentFQ).to eq cmd_hostname
+  end
+
+  it "reads /etc/hostname when hostname --fqdn fails" do
+    allow_execute_hostname(nil)
```

Comment:
and here it shows. I worry it is not common fail of reading, as it should fail with exit code and some stderr output.